pkg/envoy/cds/cluster.go

```diff
@@ -97,21 +96,21 @@ func getLocalServiceCluster(catalog catalog.MeshCataloger, proxyServiceName serv
 		Http2ProtocolOptions: &xds_core.Http2ProtocolOptions{},
 	}
 
-	endpoints, err := catalog.ListEndpointsForService(proxyServiceName)
+	ports, err := catalog.GetPortToProtocolMappingForService(proxyServiceName)
```

Comment:
Ideally we need to be using `service.spec.ports[].targetPort` here. We can only use `service.spec.ports[].port` if it is the same as the `targetPort`. Since we are attempting to create local clusters for the root service in a traffic split.

I would change this to the following:
```
if cluster == root service in split:
  ports = GetPortToProtocolMappingForService
else:
  ports = GetTargetPortToProtocolMappingForService
```

This will ensure that we aren't setting up the local clusters with incorrect port numbers for cases where the targetPort is different from the port for a service.